File: ih_patient/src/main/java/com/keydom/ih_patient/fragment/BaseFragment.java
package com.keydom.ih_patient.fragment;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * @Name：com.kentra.yxyz.fragment
 * @Description：基础Fragment 初始化操作等
 * @Author：song
 * @Date：18/11/2 下午1:42
 * 修改人：xusong
 * 修改时间：18/11/2 下午1:42
 */
public class BaseFragment extends Fragment{
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    }

}

File: ih_common/src/main/java/com/keydom/ih_common/base/BaseFragment.kt
package com.keydom.ih_common.base

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import butterknife.ButterKnife
import butterknife.Unbinder
import com.ganxin.library.LoadDataLayout
import com.keydom.ih_common.R

abstract class BaseFragment : Fragment(), BaseFragmentInterFace {
    private var loadDataLayout: LoadDataLayout? = null
    protected var isViewInitiated: Boolean = false
    protected var isVisibleToUser: Boolean = false
    protected var isDataInitiated: Boolean = false
    /**
     * 注解
     */
    protected var unbinder: Unbinder? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val inflate = inflater.inflate(layoutRes, container, false)
        getView(inflate)

        unbinder = ButterKnife.bind(this, inflate)
        isViewInitiated = true;
        loadDataLayout = inflate.findViewById<View>(R.id.loadDataLayout) as LoadDataLayout?
        return inflate
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initController()
        initData(savedInstanceState)

        if (isVisibleToUser) {
            if (!isDataInitiated && isViewInitiated) {
                lazyLoad()
                isDataInitiated = true
            }
        }
    }

    override fun setUserVisibleHint(isVisibleToUser: Boolean) {
        super.setUserVisibleHint(isVisibleToUser)
        this.isVisibleToUser = isVisibleToUser;
        if (isVisibleToUser) {
            if (!isDataInitiated && isViewInitiated) {
                lazyLoad()
                isDataInitiated = true
            }
        }
    }

    fun pageLoading() {
        loadDataLayout?.let { it.setStatus(LoadDataLayout.LOADING) }
    }

    fun pageLoadingSuccess() {
        loadDataLayout?.let { it.setStatus(LoadDataLayout.SUCCESS) }
    }

    fun pageLoadingFail() {
        loadDataLayout?.let { it.setStatus(LoadDataLayout.ERROR) }

    }

    fun pageEmpty() {
        loadDataLayout?.let { it.setStatus(LoadDataLayout.EMPTY) }
    }

    fun noNetWork() {
        loadDataLayout?.let { it.setStatus(LoadDataLayout.NO_NETWORK) }
    }


    fun setReloadListener(listener: LoadDataLayout.OnReloadListener) {
        loadDataLayout?.let { it.setOnReloadListener(listener) }

    }

    override fun onDestroy() {
        super.onDestroy()
        unbinder?.unbind()
    }

}